Firestore/core/src/nanopb/message.h

```diff
@@ -186,6 +187,49 @@ class Message {
   T proto_{};
 };
 
+/**
+ * A wrapper of Message objects that facilitates shared ownership of non-mutable
+ * Protobuf data.
+ */
+template <typename T>
+class SharedMessage {
+ public:
+  /**
+   * Creates a valid `SharedMessage` that wraps a value-constructed ("zeroed
```

Comment:
Hmm, this doesn't seem to be the case -- `message_` has no initializer, so it seems like a default-constructed `Message` would have a null pointer to a proto.